amplify/backend/function/risestandstrongcb34053a/src/routes/announcements.js

```diff
@@ -129,8 +131,19 @@ router.post('/', async (req, res) => {
  *         "404":
  *            description: Announcement with the specified announcementId does not exist
  */
+
 router.put('/:announcementId', async (req, res) => {
-    res.end();
+    const data = req.body;
+    if (!AWS_Auth.Auth.currentAuthenticatedUser()){
+        res.status(403)
+    }
+     else{
+        const recieved = await putAnnouncement(data);
+        if (typeof recieved === undefined){
+            res.status(404);
+        }
```

Comment:
Two things, you can just say if (!received). undefined is considered falsey in JavaScript meaning it's interpreted as false in a boolean context. The other thing is res.status() only sets the status of the response to 404. You need to end the response as well.